### PR TITLE
Fix ambient doc tests

### LINKED ISSUE #15498: fix flaky test in /ambient/usage/extend-waypoint-wasm
Common failure in the `ambient` test profile ([example](https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/15474/doc.test.profile-ambient_istio.io/1818217274054545408/artifacts/tests-setup-profile-ambient-a93/TestDocs/ambient/usage/extend-waypoint-wasm/test.sh/test.sh/_test_context/test.sh_output.txt)):

```
gateway.gateway.networking.k8s.io/bookinfo-gateway created
httproute.gateway.networking.k8s.io/bookinfo created
Error from server (NotFound): deployments.apps "bookinfo-gateway-istio" not found
Failed rollout of deployment bookinfo-gateway-istio in namespace default
```

This test imports a lot of snips from getting started, which it should not do.  If those are useful and will be used more, they should be moved to boilerplate snips.

### PR DESCRIPTION
**Pull request by craigbox**
The "Extend waypoints with WebAssembly plugins" docs page had a flake where it would check for the `Deployment` created by a `Gateway` immediately after the gateway was created.  It also depended on tests from the Getting Started guide, which has bitten us at least once.

I've refactored the tests to fix both problems, and updated the test docs to mention the `ambient` profile.

Further work may include setting up some helper scripts for labelling a namespace as ambient and cleaning up afterwards, but lots of the sidecar tests embed `kubectl` and `istioctl` as I have done here so we have precedent that we can iterate on later.

Closes #15498.

### CODE REVIEW

**Comment by craigbox** [comment]
/retest

Don't think you fixed authz-tcp sorry @bleggett

**Comment by bleggett** [comment]
> /retest
> 
> Don't think you fixed authz-tcp sorry @bleggett


I definitely  didn't fix it, but I thought I had hacked around it, oh well.

**Comment by bleggett** [comment]
Opened https://github.com/istio/istio.io/issues/15505

File: content/en/docs/ambient/usage/extend-waypoint-wasm/index.md [hunk 1]
```diff
@@ -63,51 +70,51 @@ The Istio agent will interpret the WasmPlugin configuration, download remote Was
...
+1. Test `/productpage` with the credentials configured in the WasmPlugin resource:
 
-    {{< text bash >}}
+    {{< text syntax=bash snip_id=test_gateway_productpage_with_credentials >}}
     $ kubectl exec deploy/sleep -- curl -s -o /dev/null -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" -w "%{http_code}" "http://bookinfo-gateway-istio.default.svc.cluster.local/productpage"
     200
     {{< /text >}}
 
-## Apply Wasm Configuration at Waypoint Proxy
+## At a waypoint, for all services
```
Comment (bleggett) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/index.md:87]:
```suggestion
## At a waypoint, for all services in a namespace
```

**Review by bleggett — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

File: content/en/docs/ambient/usage/extend-waypoint-wasm/index.md [hunk 2]
```diff
@@ -13,27 +13,34 @@ status: Alpha
 Istio provides the ability to [extend its functionality using WebAssembly (Wasm)](/docs/concepts/wasm/).
 One of the key advantages of Wasm extensibility is that extensions can be loaded dynamically at runtime. This document outlines how to extend ambient mode within Istio with Wasm features. In ambient mode, Wasm configuration must be applied to the waypoint proxy deployed in each namespace.
 
-## Install Ambient Mode and deploy test applications
+## Before you begin
 
-Follow the [Ambient Getting Started Guide](/docs/ambient/getting-started) to install Istio in ambient mode. Deploy the [sample applications](/docs/ambient/getting-started/deploy-sample-app) required for exploring waypoint proxy extensibility via Wasm. Make sure to [add the sample applications](/docs/ambient/getting-started/secure-and-visualize) to the mesh before proceeding further.
+1. Setup Istio by following the instructions in the [ambient mode Getting Started guide](/docs/ambient/getting-started).
+1. Deploy the [Bookinfo sample application](/docs/ambient/getting-started/deploy-sample-app).
+1. [Add the default namespace to the ambient mesh](/docs/ambient/getting-started/secure-and-visualize) to the mesh.
```
Comment (dhawton) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/index.md:20]:
```suggestion
1. [Add the default namespace to the ambient mesh](/docs/ambient/getting-started/secure-and-visualize).
```

**Review by dhawton — COMMENTED** [review]
1 small nit

**Comment by craigbox** [comment]
> 1 small nit

grrr, I remember seeing that

**Review by dhawton — APPROVED** [review]

**Pull request merged**